terraform/cloud-platform-components/velero.tf

```diff
@@ -0,0 +1,100 @@
+resource "kubernetes_namespace" "velero" {
+  metadata {
+    name = "velero"
+
+    labels = {
+      "component" = "velero"
+    }
+
+    annotations = {
+      "cloud-platform.justice.gov.uk/application"   = "Velero"
+      "cloud-platform.justice.gov.uk/business-unit" = "cloud-platform"
+      "cloud-platform.justice.gov.uk/owner"         = "Cloud Platform: platforms@digital.justice.gov.uk"
+      "cloud-platform.justice.gov.uk/source-code"   = "https://github.com/ministryofjustice/cloud-platform-infrastructure"
+      "iam.amazonaws.com/permitted"                 = ".*"
+    }
+  }
+}
+
+data "template_file" "velero" {
+  template = file("./templates/velero/velero.yaml.tpl")
+  vars = {
+    cluster_name = terraform.workspace
+    iam_role     = aws_iam_role.velero.name
+  }
+}
+
+data "aws_iam_policy_document" "velero_assume" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_iam_role.nodes.arn]
+    }
+  }
+}
+
+resource "aws_iam_role" "velero" {
+  name               = "velero.${data.terraform_remote_state.cluster.outputs.cluster_domain_name}"
+  assume_role_policy = data.aws_iam_policy_document.velero_assume.json
+}
+
+data "aws_iam_policy_document" "velero" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+      "ec2:DescribeSnapshots",
+      "ec2:DescribeVolumes",
+      "ec2:CreateTags",
+      "ec2:CreateVolume",
+      "ec2:CreateSnapshot",
+      "ec2:DeleteSnapshot",
+    ]
+    resources = ["*"]
```

Comment:
Is there any way to restrict this further, so that velero only has permissions over its own snapshots?